cmd/kubeadm/app/phases/addons/proxy/manifests.go

```diff
@@ -108,7 +108,16 @@ spec:
       - key: CriticalAddonsOnly
         operator: Exists
       - operator: Exists
-      nodeSelector:
-        beta.kubernetes.io/arch: {{ .Arch }}
+      affinity:
```

Comment:
... Why is the previous node-selector inconsistent?  

Affinity is super expensive. 

/cc @bsalamat

Reply:
@timothysc sorry for the late `/hold` label - we're internally evaluating changes to keep the behavior consistent. Most probably we will go that way, and then we don't need this change and will close this pr.

Reply:
@timothysc as @Huang-Wei said already, we probably won't need this change. We will fix the code on the DS controller side so that the old configuration keeps working.
BTW, inter-pod affinity is expensive. Node affinity which is used here is not that expensive.

Reply:
And long term we're hoping we can completely ditch this so we can just use manifest lists.
cc @dims @neolit123 @mkumatag